src/net_processing.cpp

```diff
@@ -1741,7 +1741,10 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
                 // See https://github.com/bitcoin/bitcoin/issues/8279 for details.
                 assert(recentRejects);
                 recentRejects->insert(tx.GetHash());
-            }
+                if (RecursiveDynamicUsage(*ptx) < 100000)
```

Comment:
A quick examination shows 80-90% of transactions not accepted to the mempool are due to rate limited free transaction.   What do you think about also adding to this check and the one below `&& state.GetRejectCode() != REJECT_INSUFFICIENTFEE`

You could be throwing away some transactions that would help blocks reconstruct, but I get close to 100 "not accepted" transactions an hour and it seems like I'd rather have 5 hours of non-free transactions rather than 1 hour of all.